scripts/functions/ae.py

```diff
@@ -0,0 +1,266 @@
+
+# The below is necessary for starting core Python generated random numbers
+# in a well-defined state.
+
+rn.seed(12345)
+
+# Force TensorFlow to use single thread.
+# Multiple threads are a potential source of
+# non-reproducible results.
+# For further details, see: https://stackoverflow.com/questions/42022950/which-seeds-have-to-be-set-where-to-realize-100-reproducibility-of-training-res
+
+session_conf = tf.ConfigProto(intra_op_parallelism_threads=1, inter_op_parallelism_threads=1)
+
+from keras import backend as K
+
+# The below tf.set_random_seed() will make random number generation
+# in the TensorFlow backend have a well-defined initial state.
+# For further details, see: https://www.tensorflow.org/api_docs/python/tf/set_random_seed
+
+tf.set_random_seed(1234)
+
+sess = tf.Session(graph=tf.get_default_graph(), config=session_conf)
+K.set_session(sess)
+
+
+from keras.layers import Input, Dense, Lambda, Layer, Activation
+from keras.layers.normalization import BatchNormalization
+from keras.models import Model, Sequential
+from keras import metrics, optimizers
+from keras.callbacks import Callback
+
+from helper_ae import sampling, CustomVariationalLayer, WarmUpCallback
+
+def ae_2layer_model(learning_rate, batch_size, epochs, kappa, intermediate_dim,
+                    latent_dim, epsilon_std, base_dir, analysis_name):
+    """
+    Train 2-layer Tybalt model using input dataset
+    
+    Output:
+     Encoding and decoding neural networks to use in downstream analysis
+    """
+    
+    # --------------------------------------------------------------------------------------------------------------------
+    # Files
+    # --------------------------------------------------------------------------------------------------------------------
+    data_file =  os.path.join(base_dir, "data", analysis_name, "train_model_input.txt.xz")
+    rnaseq = pd.read_table(data_file, index_col=0, header=0, compression='xz')
+
+    # --------------------------------------------------------------------------------------------------------------------
+    # Initialize hyper parameters
+    #
+    # learning rate: 
+    # batch size: Total number of training examples present in a single batch
+    #             Iterations is the number of batches needed to complete one epoch
+    # epochs: One Epoch is when an ENTIRE dataset is passed forward and backward through the neural network only ONCE
+    # kappa: warmup
+    # original dim: dimensions of the raw data
+    # latent dim: dimensiosn of the latent space (fixed by the user)
+    #   Note: intrinsic latent space dimension unknown
+    # epsilon std: 
+    # beta: Threshold value for ReLU?
+    # --------------------------------------------------------------------------------------------------------------------
+
+    original_dim = rnaseq.shape[1]
+    beta = K.variable(0)
+
+    stat_file =  os.path.join(base_dir, "stats", analysis_name, "ae_2layer_{}latent_stats.tsv".format(latent_dim))
```

Comment:
I wasn't when I originally wrote this.  I used to use spyder as an editor and then when I joined I started using jupyter notebooks since that seemed to be the norm, but not as familiar with its functionality so I just assumed it had the same sort of formatting for py and ipynb files.  I have learned that it doesn't:) So now using sublime.  Not sure if you have another suggestion